x-pack/plugins/endpoint/public/applications/endpoint/view/formatted_date_time.tsx

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { FormattedDate, FormattedTime, FormattedRelative } from '@kbn/i18n/react';
+
+export const FormattedDateAndTime: React.FC<{ date: Date }> = ({ date }) => {
+  // If date is greater than or equal to 24h (ago), then show it as a date
+  // else, show it as relative to "now"
+  return Date.now() - date.getTime() >= 8.64e7 ? (
```

Comment:
FYI - one of the outstanding tasks from when I originally implemented this is to change so that relative date will only be displayed if within the last `1h` (instead of last `24h`)